src/main/java/com/mongodb/hibernate/jdbc/AbstractMongoConnection.java

```diff
@@ -0,0 +1,167 @@
+    }
+
+    @Override
+    public Savepoint setSavepoint() {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public Savepoint setSavepoint(String name) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public void rollback(Savepoint savepoint) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public void releaseSavepoint(Savepoint savepoint) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public Statement createStatement(int resultSetType, int resultSetConcurrency, int resultSetHoldability) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(
+            String sql, int resultSetType, int resultSetConcurrency, int resultSetHoldability) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public CallableStatement prepareCall(
+            String sql, int resultSetType, int resultSetConcurrency, int resultSetHoldability) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String sql, int[] columnIndexes) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public boolean isValid(int timeout) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public void setClientInfo(String name, String value) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public void setClientInfo(Properties properties) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public String getClientInfo(String name) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public Properties getClientInfo() {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public void setSchema(String schema) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public String getSchema() {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public void abort(Executor executor) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public void setNetworkTimeout(Executor executor, int milliseconds) {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+
+    @Override
+    public int getNetworkTimeout() {
+        throw new IllegalStateException("Hibernate won't use this method");
+    }
+}
```

Comment:
The above abstract class contains all the `unused` API methods (in non-test code excluding delegator calling) in Hibernate v6.6 (by manual checking in IDE). All the methods really ever used by Hibernate will be left out so `MongoCollection` class needs to implement.

A common pattern for the future JDBC API implementations, as explained in the [package-info.java](https://github.com/mongodb/mongo-hibernate/pull/14/files#diff-5c1f1caf24a8bb65ec214e3bfbc8a6e4a3564946df7cc27668a301d24d2d00e6)